client/src/components/MyClubs.tsx

```diff
@@ -1,10 +1,117 @@
+import React, { useState, useEffect } from 'react';
 import NavBar from "./NavigationBar";
 
 export default function MyClubs() {
+  const initialClubs = [
+    { name: 'Tennis Club', id: 'tennis' },
+    { name: 'Chess Club', id: 'chess' },
+    { name: 'Reading Club', id: 'reading' },
+    { name: 'Science Club', id: 'science' }
+  ];
+
+  const [clubs, setClubs] = useState(initialClubs);
+  const [leftClubs, setLeftClubs] = useState([]);
+  const [query, setQuery] = useState('');
+
+  // Function to handle leaving a club
+  const leaveClub = (clubId) => {
+    const leftClub = clubs.find(club => club.id === clubId);
+    setClubs(clubs.filter(club => club.id !== clubId));
+    setLeftClubs([...leftClubs, leftClub]);
+
+    // Set a timer to automatically remove the undo option after 30 seconds
+    setTimeout(() => {
+      setLeftClubs(leftClubs => leftClubs.filter(club => club.id !== clubId));
+    }, 5000); // 30 seconds
+  };
+
+  // Function to handle undoing leave action
+  const undoLeave = (clubId) => {
+    const returnedClub = leftClubs.find(club => club.id === clubId);
+    setClubs([...clubs, returnedClub]);
+    setLeftClubs(leftClubs.filter(club => club.id !== clubId));
+  };
+
+  // Filter clubs based on query
+  const filteredClubs = clubs.filter(club =>
+    club.name.toLowerCase().includes(query.toLowerCase())
+  );
+
   return (
-    <div>
+    <div style={{ textAlign: 'center' }}>
```

Comment:
Can you move all in-line style instances like this to a CSS file? @tmohaupt has [an example of moving styles to a CSS file here](https://github.com/varangill/ClubHub/pull/9/commits/7d70e4a6c0551b49dd66e341acd208e4e6e3c6fc) if you need help

There's also in-line styles on lines 45, 49, 61, 65, and the three style constants at the bottom of the file that could be removed to clean up the file a bit

Reply:
I entered the changes.